openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -191,6 +199,141 @@ def update_account_settings(requesting_user, update, username=None):
+    extended_profile_fields_data: dict, user, field_errors: dict
+) -> Optional[forms.Form]:
+    """
+    Get or create an extended profile form instance.
+
+    Attempts to create a form instance using the configured `REGISTRATION_EXTENSION_FORM`.
+    If an extended profile model exists, tries to bind to existing user data or creates
+    a new instance. Handles import errors and missing configurations gracefully.
+
+    Args:
+        extended_profile_fields_data (dict): Extended profile field data to populate the form
+        user (User): User instance to associate with the extended profile
+        field_errors (dict): Dictionary to collect validation errors if form creation fails
+
+    Returns:
+        Optional[forms.Form]: Extended profile form instance with user data, or None if
+        no extended profile form is configured or creation fails
+    """
+    try:
+        extended_profile_model = get_extended_profile_model()
+
+        kwargs = {}
+        if not extended_profile_model:
+            logger.info("No extended profile model configured")
+        else:
+            try:
+                kwargs["instance"] = extended_profile_model.objects.get(user=user)
+            except ObjectDoesNotExist:
+                logger.info("No existing extended profile found for user %s, creating new instance", user.username)
+
+        extended_profile_form = get_registration_extension_form(data=extended_profile_fields_data, **kwargs)
+
+        return extended_profile_form
+
+    except ImportError as e:
+        logger.warning("Extended profile model not available: %s", str(e))
+        return None
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        logger.error("Unexpected error creating custom form for user %s: %s", user.username, str(e))
+        field_errors["extended_profile"] = {
+            "developer_message": f"Error creating custom form: {str(e)}",
+            "user_message": _("There was an error processing the extended profile information"),
+        }
+        return None
+
+
+def _validate_extended_profile_form_and_collect_errors(extended_profile_form: forms.Form, field_errors: dict) -> None:
+    """
+    Validate the extended profile form and collect any validation errors.
+
+    Args:
+        extended_profile_form (forms.Form): The extended profile form to validate
+        field_errors (dict): Dictionary to collect validation errors
+    """
+    if not extended_profile_form.is_valid():
+        logger.info("Extended profile form validation failed with errors: %s", extended_profile_form.errors)
+
+        for field_name, field_errors_list in extended_profile_form.errors.items():
+            first_error = field_errors_list[0] if field_errors_list else "Unknown error"
+
+            field_errors[field_name] = {
+                "developer_message": f"Error in extended profile field {field_name}: {first_error}",
+                "user_message": str(first_error),
+            }
```

Comment:
I think we can reduce 1 level of nestiness here by:
```suggestion
    if extended_profile_form.is_valid():
        return # Nothing to do here, the form is valid

    logger.info("Extended profile form validation failed with errors: %s", extended_profile_form.errors)

    for field_name, field_errors_list in extended_profile_form.errors.items():
        first_error = field_errors_list[0] if field_errors_list else "Unknown error"

        field_errors[field_name] = {
            "developer_message": f"Error in extended profile field {field_name}: {first_error}",
            "user_message": str(first_error),
        }
```

Reply:
Done: https://github.com/openedx/edx-platform/pull/37119/commits/db4818cdbd061d613b0fbeb4a5449064354f3057, thanks!